[Jerry Lan] iP

Lan-Jingbo opened the pull request:
# Duke

- 下午上课
- 下午买奶茶

1. rua Ashy
2. rua Putu

Here are some sample code for `print` function different languages
```ruby
>>> print("Hello CS2103T and CS2101")
>>> System.out.println("Hello CS3230")
```

- [x] CS2103T
- [x] CS3230
- [x] PC2411
- [ ] CS2106

互联网真的是太有意思啦:sweat_smile:

> I can do it -- Prof Aaron Tan

**Learn** ~~CS2030S~~ *CS2103T* from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)

Commit:
Add Gradle support

Lan-Jingbo committed:
Update Duke.java

Lan-Jingbo committed:
Update Duke.java

The second change: fix problem of cannot scan all line

Lan-Jingbo committed:
reserve for tmr

Lan-Jingbo committed:
Update for Level 2 and 3

Lan-Jingbo committed:
Completion of Level 3/4

Lan-Jingbo committed:
Final Draft(Complete all)

Lan-Jingbo committed:
change

Lan-Jingbo committed:
Level-7 Complete

Lan-Jingbo committed:
Level-8-Draft

Lan-Jingbo committed:
Merge branch 'branch-Level-8'

Lan-Jingbo committed:
A small update for level8

Lan-Jingbo committed:
Level-8

Lan-Jingbo committed:
Merge branch 'add-gradle-support'

Lan-Jingbo committed:
Addsomething

Lan-Jingbo committed:
?

Lan-Jingbo committed:
Revert "?"

This reverts commit d084f6913d2b2cbf203b29faf8b15c892f44f47d.

Lan-Jingbo committed:
Merge branch 'branch-Level-8'

Lan-Jingbo committed:
A-MoreOOP

Lan-Jingbo committed:
A-Package

Lan-Jingbo committed:
A-JUnit

Lan-Jingbo committed:
A-JavaDoc

Lan-Jingbo committed:
Level-9

Lan-Jingbo committed:
Level-9 update

Lan-Jingbo committed:
A-CodingStandard

Lan-Jingbo committed:
Merge branch 'branch-A-CodingStandard'

nealetham commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,188 @@
+            try {
+                if (str.startsWith("todo")) {
+                    tasks.addTask(Parser.todo(str));
+                    System.out.println("Now you have " + tasks.getSize() + " task(s)");
```
Perhaps you can abstract this line into another method? So that there is less repetitive lines in your code.

nealetham commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,70 @@
+ * Event is a kind of Task.
+ * @author Lan Jingbo, Jerry
+ */
+class Event extends Task{
```
Remember to put a space after Task. i.e. 
```suggestion
class Event extends Task {
```

jq1836 replied:
As mentioned by Neale, do remember to put a space after Task!

nealetham commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+     * @return The tasks in the file
+     * @throws WrongMessageException the potential exceptions
+     */
+    public ArrayList<Task> extractFile() throws WrongMessageException {
```
Very readable and intuitive! 👍

nealetham commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,46 @@
+        return isDone ? "Y" : "N";
+    }
+
+    public void donelah() {
```
Perhaps you can use camelCase for this method name? Generally, the naming convention for setter methods are setX().

nealetham reviewed (commented):
Overall, I found your code easy to read for the most part.  I noted a few minor coding standard violations too. Some of the classes are getting quite long (Duke.java). Consider splitting into
smaller classes if that makes sense.

jq1836 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,188 @@
+package duke;
+
+import java.util.List;
+import java.io.*;
```
Try to avoid inefficient wildcard imports. Importing all used classes explicitly helps make your code clearer!

jq1836 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    /**
+     * The method for saving tasks into file.
```
Hey this might be a mistake, really shows how readable your code is such that even with the wrong Java Doc, its still possible to understand what your method does. Keep this up!

jq1836 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,46 @@
+        isDone = true;
+    }
+
+    public void nodone() {
```
Similar to what Neale mentioned with your donelah method! nodone might suggest something else!

jq1836 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,46 @@
+        return false;
+    }
+
+    public boolean hasThis(String str) {
```
Your hasThis method might be misleading. Perhaps a better name would be hasSubstring or hasThisString!

jq1836 commented on src/test/java/duke/DukeTest.java:
```diff
@@ -0,0 +1,16 @@
+package duke;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
I'm unsure about this, but this might violate the mod's coding standards. This should go right after the package if I'm not wrong. Try adding the mod's checkstyle file, it helps detect things like this!

jq1836 reviewed (commented):
I noticed that your Java Docs is incomplete. Try to finish it up ASAP! It really makes your code much more readable! Code is really neat and Java Docs enhances the readability of your code! Minimal Java Coding Standards violations, good job! Try to organize your classes into relevant packages, really helps neaten up your project! Overall, great job!

Lan-Jingbo committed:
nothing here

Lan-Jingbo committed:
nothing here-2

Lan-Jingbo committed:
basic-Level-10

LimAiLin commented on src/main/java/Duke.java:
```diff
@@ -12,23 +12,51 @@ public static void main(String[] args) {
+                    break;
+
+                case "event":
+                    String eventstr = sc.nextLine();
```
Your variable names should be in camelCase following the Java coding standard.

LimAiLin commented on src/main/java/Task.java:
```diff
@@ -8,7 +8,7 @@ public Task(String itself) {
+        return isDone ? "Y" : "N";
     }
 
     public void donelah() {
```
Strange way of naming your methods, method name should be in camelCase too.

Lan-Jingbo committed:
New Things

Lan-Jingbo committed:
Added assertions to the code

The duke program classes have some common behaviors.
The common behaviors across Storage class cause code duplication and cannot pretend potential errors.

Let's,
* Add assertions executions at IOException clause
* Analyze the size of task list after start the program

Lan-Jingbo committed:
small update

Lan-Jingbo committed:
Merge pull request #1 from Lan-Jingbo/branch-A-assertions

Branch a assertions

Lan-Jingbo committed:
A re-write of UI

The previous Ui is too simple,
so I want to make it more functions.
I changed the Parser class to ensure all commands can return a String type.
Completed

Lan-Jingbo committed:
Improve code quality

Lan-Jingbo committed:
Merge pull request #3 from Lan-Jingbo/branch-A-CodeQuality

Improve code quality

Lan-Jingbo committed:
Add help function

Lan-Jingbo committed:
Merge pull request #4 from Lan-Jingbo/C-Help

Add help function

Lan-Jingbo committed:
Update README.md

Lan-Jingbo committed:
A-UI Screenshot

Lan-Jingbo committed:
Update on README.md

Lan-Jingbo committed:
Change in the exit delay time

From 0 seconds to 4 seconds, which will be much more smooth to use.

Lan-Jingbo committed:
Create MANIFEST.MF

Lan-Jingbo committed:
Update build.gradle

the previous mainClassName was wrong

Lan-Jingbo committed:
Small Update

Lan-Jingbo committed:
Update Ui.png

Lan-Jingbo committed:
Update Main.java

Lan-Jingbo committed:
test

Lan-Jingbo committed:
hi

Lan-Jingbo committed:
Update on Storage
